#
# When an entry is removed, we need to check the one that comes in its
# place, if not, we might even skip across the NULL and crash.
#
--- xterm-384/main.c	2024-01-09 20:09:24.609195665 +0100
+++ xterm-384/main.c	2024-01-09 20:11:55.911118800 +0100
@@ -3786,13 +3786,18 @@
 			my_var[dstlen] = '\0';
 			xtermUnsetenv(my_var);
 			free(my_var);
+			/* When removing an entry, check the same slot again. */
+			j--;
 		    }
 		    break;
 		}
 	    } else if (dstlen == srclen &&
 		       !strncmp(environ[j], table[k].name, srclen)) {
-		if (table[k].trim)
+		if (table[k].trim) {
 		    xtermUnsetenv(table[k].name);
+		    /* When removing an entry, check the same slot again. */
+		    j--;
+		}
 		break;
 	    }
 	}
